dm/relay/binlog_writer.go

```diff
@@ -105,32 +176,46 @@ func (w *BinlogWriter) Close() error {
 
 	w.file = nil
 	w.offset.Store(0)
-	w.uuid = ""
-	w.filename = ""
+	w.uuid.Store("")
+	w.filename.Store("")
+	w.input = nil
 
+	if writeErr := w.Error(); writeErr != nil {
+		return writeErr
+	}
 	return err
 }
 
 func (w *BinlogWriter) Write(rawData []byte) error {
-	w.mu.RLock()
-	defer w.mu.RUnlock()
-
 	if w.file == nil {
 		return terror.ErrRelayWriterNotOpened.Delegate(errors.New("file not opened"))
 	}
+	w.input <- rawData
+	w.offset.Add(int64(len(rawData)))
```

Comment:
`BinlogReader` will use this offset to check whether there's need to re-parse binlog, maybe move it after we actual write data.

Reply:
we use offset to check whether hole exist, so cannot update it later.

Reply:
https://github.com/pingcap/tiflow/blob/d98f885b51eb8d3cb35a960598dbe025281cf733/dm/relay/relay_writer.go#L304

Reply:
use another offset?